rfcs/59-writemultipart-returns-part.md

```diff
@@ -0,0 +1,64 @@
+---
+author: JinnyYi <github.com/JinnyYi>
+status: draft
+updated_at: 2021-05-11
+---
+
+# AOS-59: WriteMultipart Returns Part
+
+## Background
+
+`Multiparter` is designed for multipart upload. Multipart upload is a three-step process: use `CreateMultipart` to initiate the upload, use `WriteMultipart` to upload the object parts, and after you have uploaded all the parts, use `CompleteMultipart` to complete the multipart upload. You can use `ListMultipart` to list all of your in-progress multipart uploads or get a list of the parts that you have uploaded for a specific multipart upload.
+
+`CompleteMultipart` request must include the upload ID and a list of both part numbers and corresponding ETag values returned after that part was uploaded in some services. The ETag uniquely identifies the combined object data, not necessarily an MD5 hash of the object data. We need return ETag that we got from services to make it possible. 
+
+## Proposal
+
+So I propose to change `WriteMultipart` to: 
+
+```go
+WriteMultipart(o *Object, r io.Reader, size int64, index int, pairs ...Pair) (n int 64, p Part, err error)

```

Comment:
`Part` should be  `*Part`, keep the same with `CompleteMultipart`

Or, we should propose to change `CompleteMultipart` either.